### DISCUSSION TITLE
Use TypeScript imports resolver

### DISCUSSION THREAD

**Discussion by mrhyde** [standard/eslint-config-standard-with-typescript#1952] [Ideas]
Hi! Are there any plans to use [TypeScript imports resolver](https://github.com/import-js/eslint-plugin-import?tab=readme-ov-file#typescript) in `eslint-plugin-import`? It seems a bit off that this package is designed for TypeScript but doesn't include this by default.

P.S. Thank you for all your hard work over the years! Your efforts deserve more recognition!

**Comment by mightyiam** [comment]
Thank you. This seems to introduce a good amount of details. Would you be interested in contributing it?

**Comment by mrhyde** [comment]
Sure! Just wanted to be sure that it's something you are interested in

**Comment by mrhyde** [comment]
@mightyiam I have created https://github.com/mightyiam/eslint-config-love/pull/1509 let me know what you think

**Comment by psychobolt** [comment]
Seems like the original PR was rejected. In my case, I'm not looking for a code mod feature, but somehow VSCode doesn't really provide great visibility for unused imports e.g.

<img width="821" alt="Screenshot 2024-07-09 at 9 05 40 PM" src="https://github.com/mightyiam/eslint-config-love/assets/560721/0a06c26a-dcc3-4b2e-b733-b468879f3e05">

As you can see, lower opacity is something not ideal. However, if I however over the the problem, I can see it clearly.

<img width="1018" alt="Screenshot 2024-07-09 at 9 05 51 PM" src="https://github.com/mightyiam/eslint-config-love/assets/560721/168247a0-c6cd-4feb-b21c-1bb82a34930c">

TS Language Server also doesn't seem to populate the Problems tab and my TypeScript compilier doesn't even complain about it! So my project will probably be filled with unused code that requires code modding. Not what I expected if we are able to do a lint checker. I also expected TS to not give out false positives...

Anyway for alternatives, I suggest:
https://github.com/un-ts/eslint-plugin-import-x
https://www.npmjs.com/package/eslint-import-resolver-typescript
https://www.npmjs.com/package/eslint-plugin-unused-imports

**Comment by mightyiam** [comment]
@psychobolt doesn't that seem like an IDE issue to you? Does `tsc` complain about it?

**Comment by psychobolt** [comment]
Seems like you have to explicitly tell TypeScript to warn about unused variables e.g
```json
    "noUnusedLocals": true
```

![Screenshot 2024-07-10 at 7 55 34 AM](https://github.com/mightyiam/eslint-config-love/assets/560721/1e79ad26-cd81-4533-9364-0f2302841743) 

If I turn on this compilerOption, tsc will complain. Anyways there is already a rule `no-unused-vars` for ts-eslint: https://typescript-eslint.io/rules/no-unused-vars/ which seem enabled by default. I'm not sure if import resolver is suppose to handle that, but eslint is not warning about unused imports. Until I enabled that option I mentioned, those TS errors will be shown also in linter.

Some more info on how the IDE works: https://github.com/microsoft/vscode/issues/219335
